### DISCUSSION TITLE
Is hardware/software flow control working on OTBR?

### DISCUSSION THREAD

**Discussion by EricSTMicro** [openthread/openthread#10054] [Q&A]
Hello,

I'm working on the OTBR, using a RPi 4B and a STM32WB55.

Due to spinel/HDLC over UART issues, leading to reset of the Thread application on the Host, I would like to make it more robust.

To enable it on Host, I made the following changes in **/etc/default/otbr-agent**:
`OTBR_AGENT_OPTS="-I wpan0 -B wlan0 spinel+hdlc+uart:///dev/ttyACM0?uart-baudrate=460800&uart-flow-control trel://wlan0"`

But it doesn't seem to have any effect.

Different setups have been tried:

- USB cable:
  - by connecting directly from the STM32 USB port (from the ST-Link using a VCP) to the RPi's USB port. I'm not sure how this is supposed to work, as the RTS/CTS signals are not "linked" to the USB. I assume we need to activate some kind of XON/XOFF for software flow control on the STM32. I don't know if this is possible.
  
- FTDI serial/USB converter cable:
  - by connecting the needed signals (TX/RX/GND/RTS/CTS) to the FTDI cable, which is connected to the RPi's USB port on the other side. I assumed this was the best way to make it work, but it doesn't. By checking with a logic analyzer, I could see the RPi doesn't take the CTS/RTS signals into account and keep sending data, regardless of their state.
  
- Pin to pin:
  - by connecting the needed signals (TX/RX/GND/RTS/CTS) directly from STM32's GPIO port to RPi's GPIO port. Spinel overt UART communication is working well, but as above, the RTS/CTS signals are not taking into account by the RPi.
  

Questions:
- Is the flow control a hardware of a software one?
- What is the correct way to use it?

Regards,
Eric

**Comment by jwhui** [comment]
The termios configuration happens here:

https://github.com/openthread/openthread/blob/e27885ecb4b745fb819a2e311b37fda3bd79248f/src/posix/platform/hdlc_interface.cpp#L601

Please feel free to propose alternative solutions. Contributions are always welcome!

**Comment by vanthome** [comment]
@EricSTMicro I have the exact same issue with Baud Rate 460800. Could you make in a stable manner it work somehow?
